lms/djangoapps/courseware/views/views.py

```diff
@@ -497,10 +420,52 @@ def get(self, request, course_id, tab_type, **kwargs):
         course_key = CourseKey.from_string(course_id)
         with modulestore().bulk_operations(course_key):
             course = get_course_with_access(request.user, 'load', course_key)
-            tab = CourseTabList.get_tab_by_type(course.tabs, tab_type)
-            page_context = self.create_page_context(request, course=course, tab=tab, **kwargs)
-            set_custom_metrics_for_course_key(course_key)
-            return super(CourseTabView, self).get(request, course=course, page_context=page_context, **kwargs)
+            try:
+
+                # Verify that the user has access to the course
+                redirect_if_needed_to_access_course(request, course)
+
+                # Render the page
+                tab = CourseTabList.get_tab_by_type(course.tabs, tab_type)
+                page_context = self.create_page_context(request, course=course, tab=tab, **kwargs)
+                set_custom_metrics_for_course_key(course_key)
+                return super(CourseTabView, self).get(request, course=course, page_context=page_context, **kwargs)
+            except Exception as exception:  # pylint: disable=broad-except
+                return CourseTabView.handle_unexpected_error(request, course, exception)
+
+    @staticmethod
+    def handle_unexpected_error(request, course, exception):
+        """
+        Handle unexpected exceptions raised when rendering a view.
+        """
+        if settings.DEBUG:
```

Comment:
That's a great catch. I've moved the DEBUG clause down so that it is only unexpected exceptions that are thrown and not logged.